File: back/src/main/java/com/p_avanzada/taller/models/Orden.java
package com.p_avanzada.taller.models;

import jakarta.persistence.Entity;
import jakarta.persistence.Table;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinTable;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Column;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import java.util.List;
import java.util.Date;

@Entity
@Table(name = "Orden", schema = "public")
public class Orden {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    @ManyToOne
    @JoinColumn(name = "tecnico_id")
    private Tecnico tecnico;

    @ManyToOne
    @JoinColumn(name = "vehiculo_patente")
    private Vehiculo vehiculo;

    @ManyToMany
    @JoinTable(name = "OrdenServicio", joinColumns = @JoinColumn(name = "orden_id"), inverseJoinColumns = @JoinColumn(name = "servicio_id"))
    private List<Servicio> servicios;

    @Column(name = "descripcion", length = 1000)
    private String descripcion;
    
    @Column(name = "estado")
    private boolean estado = true;
 
    @ManyToOne
    @JoinColumn(name = "estadoGestion")
    private EstadoGestion estadoGestion;

    @Column(name = "fecha")
    private Date fecha;


    public void delete() {
        estado = false;
    }

    public void recover() {
        estado = true;
    }
    
    public boolean getEstado() {
        return estado;
    }

    public void setEstado(boolean estado) {
        this.estado = estado;
    }

    public EstadoGestion getEstadoGestion() {
        return estadoGestion;
    }

    public void setEstadoGestion(EstadoGestion estadoGestion) {
        this.estadoGestion = estadoGestion;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public void setCliente(Cliente cliente) {
        this.cliente = cliente;
    }

    public Cliente getCliente() {
        return cliente;
    }

    public Tecnico getTecnico() {
        return tecnico;
    }
    
    public void setTecnico(Tecnico tecnico) {
        this.tecnico = tecnico;
    }
    
    public String getDescripcion() {
        return descripcion;
    }

    public Vehiculo getVehiculo() {
        return vehiculo;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public void setVehiculo(Vehiculo vehiculo) {
        this.vehiculo = vehiculo;
    }

    public List<Servicio> getServicios() {
        return servicios;
    }

    public void setServicios(List<Servicio> servicios) {
        this.servicios = servicios;
    }
    public Date getFecha() {
        return fecha;
    }
    
    public void setFecha(Date fecha) {
        this.fecha = fecha;
    }
    
}

File: front/src/Utils/Orden.js
import { postFetch, getFetch, host } from "./Fetchs";

export async function postNewOrden(cliente_id, vehiculo_patente, servicios, descripcion, tecnico_id, fecha, estadoGestion_id) {
    return postFetch(`${host}api/orden/new`, { cliente: { id: cliente_id }, vehiculo: { patente: vehiculo_patente }, servicios: servicios, descripcion: descripcion, fecha: fecha, tecnico: { id: tecnico_id }, estadoGestion: { id: estadoGestion_id } });
};

export async function getOrdenes() {
    return getFetch(`${host}api/orden/listar`)
}

export async function getOrdenesDeleted() {
    return getFetch(`${host}api/orden/listar/deleted`);
}

export async function postAlterOrden(data) {
    return postFetch(`${host}api/orden/alter`, data);
}

export async function postDeleteOrden(id) {
    return postFetch(`${host}api/orden/delete`, { id: id });
}

export async function postRecoverOrden(id) {
    return postFetch(`${host}api/orden/recover`, { id: id });
}